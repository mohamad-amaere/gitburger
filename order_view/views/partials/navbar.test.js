const fs = require("fs");
const path = require("path");
const ejs = require("ejs");
const { JSDOM } = require("jsdom");

describe("navbar.ejs", () => {
	let document;

	beforeAll(async () => {
		const filePath = path.join(__dirname, "navbar.ejs");
		const template = fs.readFileSync(filePath, "utf-8");
		const html = await ejs.render(template, {
			cart: [{ burgerId: "1", quantity: 2, price: 50 }]
		});

		const dom = new JSDOM(html, { runScripts: "dangerously" });
		document = dom.window.document;
	});

	it("should match the snapshot", () => {
		expect(document.documentElement.outerHTML).toMatchSnapshot();
	});

	it("should open the cart dialog when the cart button is clicked", () => {
		const cartButton = document.getElementById("cart-button");
		const cartDialog = document.getElementById("cart-dialog");

		expect(cartDialog.classList.contains("active")).toBe(false);

		cartButton.click();

		expect(cartDialog.classList.contains("active")).toBe(true);
	});
});
