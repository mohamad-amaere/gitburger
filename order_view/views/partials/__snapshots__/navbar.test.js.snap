// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar.ejs should match the snapshot 1`] = `
"<html><head><style>
  nav {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 10px;
    background-color: lightblue;
  }

  ul {
    display: flex;
    list-style-type: none;
  }

  li {
    margin-right: 10px;
    background-color: #f0f0f0;
    padding: 1rem .5rem;
    border-radius: 10%;
  }

  li:hover {
    background-color: #e0e0e0;
  }

  a {
    text-decoration: none;
    color: black;
  }

  div {
    display: flex;
    align-items: center;
  }

  .material-symbols-outlined {
    font-variation-settings:
      'FILL' 0,
      'wght' 400,
      'GRAD' 0,
      'opsz' 24
  }

  .amountInCart {
    margin-left: 5px;
  }

  .cart-dialog {
    display: none;
    position: absolute;
    top: 50px;
    right: 10px;
    width: 300px;
    background-color: white;
    border: 1px solid #ccc;
    box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
    padding: 10px;
    z-index: 1000;
  }

  .cart-dialog.active {
    display: block;
  }
</style>

</head><body><nav>
  <ul>
    <li><a href="/">Home</a></li>
  </ul>

  <div>
    <span class="material-symbols-outlined" id="cart-button">
      shopping_cart
    </span>
    <span class="amountInCart">1</span>
  </div>
</nav>

<div class="cart-dialog" id="cart-dialog">
  <h3>Your Cart</h3>
  <ul>
    
      <li>1 - 2</li>
    
  </ul>
  <p>total: 50 Kr
  </p><form method="post" action="/order">
    <button type="submit">Place order</button>
  </form>
</div>

<script>
  document.getElementById('cart-button').addEventListener('click', function() {
    var cartDialog = document.getElementById('cart-dialog');
    cartDialog.classList.toggle('active');
  });

  document.addEventListener('click', function(event) {
    var cartDialog = document.getElementById('cart-dialog');
    if (!cartDialog.contains(event.target) && event.target.id !== 'cart-button') {
      cartDialog.classList.remove('active');
    }
  });
</script>
</body></html>"
`;
